### PR TITLE
Add ClassName property to ResxSourceGenerator

### PR DESCRIPTION
**Pull request by raulsntos**
Allows specifying the fully-qualified class name for the class that will be generated by ResxSourceGenerator.

`System.Private.CoreLib` specifies the class name `System.SR` for a resource in `Resources\Strings.resx`:

https://github.com/dotnet/runtime/blob/8ac30d41d0be7a6de919d765e0143af865371cd4/src/coreclr/System.Private.CoreLib/System.Private.CoreLib.csproj#L32-L37

This allows doing the same with the source generator, the property uses the same name (`ClassName`):

```csproj
<EmbeddedResource Update="Resources\Strings.resx">
    <ClassName>System.SR</ClassName>
</EmbeddedResource>
```

The code for generating with a custom class name already existed, I only exposed it with the `ClassName` property.

### CODE REVIEW

**Comment by raulsntos** [comment]
@microsoft-github-policy-service agree

File: src/Microsoft.CodeAnalysis.ResxSourceGenerator/Microsoft.CodeAnalysis.ResxSourceGenerator.UnitTests/Resources/SingleString_ClassNameCSharpAsync_NS1.NS2/Resources.Designer.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+namespace TestProject
+{
+    internal static class Resources { }
+}
+namespace NS1
+{
+    internal static partial class NS2
+    {
+        private static global::System.Resources.ResourceManager? s_resourceManager;
+        public static global::System.Resources.ResourceManager ResourceManager => s_resourceManager ?? (s_resourceManager = new global::System.Resources.ResourceManager(typeof(TestProject.Resources)));
```
Comment (sharwell) [inline src/Microsoft.CodeAnalysis.ResxSourceGenerator/Microsoft.CodeAnalysis.ResxSourceGenerator.UnitTests/Resources/SingleString_ClassNameCSharpAsync_NS1.NS2/Resources.Designer.cs:15]: ❓ Is this correct? Just want to make sure.

Comment (raulsntos) [reply src/Microsoft.CodeAnalysis.ResxSourceGenerator/Microsoft.CodeAnalysis.ResxSourceGenerator.UnitTests/Resources/SingleString_ClassNameCSharpAsync_NS1.NS2/Resources.Designer.cs:15]: There's a comment that explains why this empty type is created:

https://github.com/dotnet/roslyn-analyzers/blob/7aba2caa536e168f93da3a23e427887521f57a4d/src/Microsoft.CodeAnalysis.ResxSourceGenerator/Microsoft.CodeAnalysis.ResxSourceGenerator/AbstractResxGenerator.cs#L573-L575

I'm not sure why we can't use the other type though.

**Comment by raulsntos** [comment]
Is there something else I need to do on my end?

**Comment by sharwell** [comment]
I'm not sure why the system hasn't registered your CLA acceptance. Can you try submitting another PR with something small like a test documentation change to see if it works there?

**Comment by raulsntos** [comment]
It looks like the CLA requirements are met. Is there anything else you need me to do?

**Pull request merged**